File: TOMIMSV4/Entity/Transacciones/PreFactura/Trans_Prefactura_Enc/clsBeTrans_prefactura_enc.vb
Public Class clsBeTrans_prefactura_enc
    Implements ICloneable

    Public Property IdTransPrefacturaEnc() As Integer = 0
    Public Property IdBodega() As Integer = 0
    Public Property IdPropietarioBodega() As Integer = 0
    Public Property IdClienteBodega() As Integer = 0
    Public Property IdOrdenCompraEnc() As Integer = 0
    Public Property IdTipoCuenta() As Integer = 0
    Public Property Tipo_Cambio() As Double = 0.0
    Public Property IdOrdenCompraPol() As Integer = 0
    Public Property Poliza_oc_numero_orden() As String = ""
    Public Property IdOrdenPedidoEnc() As Integer = 0
    Public Property IdOrdenPedidoPol() As Integer = 0
    Public Property Poliza_pe_numero_orden() As String = ""
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Anulada() As Boolean = False
    Public Property Fecha_desde() As Date = Date.Now
    Public Property Fecha_hasta() As Date = Date.Now
    Public Property Es_consolidador() As Boolean = False
    Public Property Observacion() As String = ""
    Public Property Procesado_erp As Boolean = False
    Public Property Autorizacion_erp As String = ""
    Public Property Cobro_peso_bruto As Boolean = False
    Public Property Variante_cobro As Boolean = False
    Public Property Agrupar_producto As Boolean = False
    Public Property Valor_Aduana As Double = 0.00
    Public Property Valor_General As Double = 0.00
    Public Property Valor_Peso As Double = 0.00

    Sub New()
        '#EJC: Add your constructor here... 
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Prefactura/clsBeTrans_prefactura_enc.cs
namespace WMS.EntityCore.Prefactura
{
    public class clsBeTrans_prefactura_enc : ICloneable
    {
        #region Propiedades

        public int IdTransPrefacturaEnc { get; set; } = 0;
        public int IdPropietarioBodega { get; set; } = 0;
        public int IdClienteBodega { get; set; } = 0;
        public int IdBodega { get; set; } = 0;
        public DateTime Fecha { get; set; } = DateTime.Now;
        public DateTime Fecha_desde { get; set; } = DateTime.Now;
        public DateTime Fecha_hasta { get; set; } = DateTime.Now;
        public double Tipo_Cambio { get; set; } = 1.0;
        public string Observacion { get; set; } = "";
        public bool procesado_erp { get; set; } = false;
        public bool anulada { get; set; } = false;
        public bool es_consolidador { get; set; } = false;
        public string poliza_oc_numero_orden { get; set; } = "";
        public string poliza_pe_numero_orden { get; set; } = "";
        public string User_agr { get; set; } = "";
        public DateTime Fec_agr { get; set; } = DateTime.Now;
        public string User_mod { get; set; } = "";
        public DateTime Fec_mod { get; set; } = DateTime.Now;
        public int IdTipoCuenta { get; set; } = 0;

        /// <summary>
        /// Indica si es un nuevo registro (para saber si insertar o actualizar)
        /// </summary>
        public bool IsNew { get; set; } = true;

        #endregion

        #region Constructores

        public clsBeTrans_prefactura_enc()
        {
        }

        public clsBeTrans_prefactura_enc(
            int idTransPrefacturaEnc,
            int idPropietarioBodega,
            int idBodega,
            DateTime fecha,
            DateTime fechaDesde,
            DateTime fechaHasta)
        {
            IdTransPrefacturaEnc = idTransPrefacturaEnc;
            IdPropietarioBodega = idPropietarioBodega;
            IdBodega = idBodega;
            Fecha = fecha;
            Fecha_desde = fechaDesde;
            Fecha_hasta = fechaHasta;
        }

        #endregion

        #region ICloneable

        public object Clone()
        {
            return this.MemberwiseClone();
        }

        #endregion
    }
}